src/java.base/share/classes/java/security/PEMDecoder.java

```diff
@@ -38,101 +38,112 @@
 import java.nio.charset.StandardCharsets;
 import java.security.cert.*;
 import java.security.spec.*;
+import java.util.Arrays;
 import java.util.Base64;
 import java.util.Objects;
 
 /**
  * {@code PEMDecoder} implements a decoder for Privacy-Enhanced Mail (PEM) data.
  * PEM is a textual encoding used to store and transfer security
  * objects, such as asymmetric keys, certificates, and certificate revocation
- * lists (CRLs).  It is defined in RFC 1421 and RFC 7468.  PEM consists of a
+ * lists (CRLs).  It is defined in RFC 1421 and RFC 7468. PEM consists of a
  * Base64-formatted binary encoding enclosed by a type-identifying header
  * and footer.
  *
- * <p> The {@linkplain #decode(String)} and {@linkplain #decode(InputStream)}
- * methods return an instance of a class that matches the data
- * type and implements {@link DEREncodable}.
+ * <p> The {@link #decode(String)} and {@link #decode(InputStream)}
+ * methods return an instance of a class that matches the PEM type and
+ * implements {@link DEREncodable}.
  *
  * <p> The following lists the supported PEM types and the {@code DEREncodable}
  * types that each are decoded as:
  * <ul>
  *  <li>CERTIFICATE : {@code X509Certificate}</li>
+ *  <li>X509 CERTIFICATE : {@code X509Certificate}</li>
+ *  <li>X.509 CERTIFICATE : {@code X509Certificate}</li>
+ *  <li>CRL : {@code X509CRL}</li>
```

Comment:
X509 CERTIFICATE, X.509 CERTIFICATE, and CRL are not standard types as defined in RFC 7468. Putting them in this list means all implementations must support them, which I don't think is appropriate. I think these types should be moved to the Implementation Note.